module-2/backend/week_one.md

```diff
@@ -1,32 +1,159 @@
+
+16. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.
+
+  ```ruby
+  get '/restaurants' do
+    @restaurants = Restaurant.all
+    erb: index
   end
   ```
 
-8. In the same code block, how would I pass a local variable `name` with a value of `Mr. Ed`?
-9. What's the purpose of ERB?
-10. Why do I need a development AND test database?
-11. What's responsive design?
-12. What is CRUD and why is it important?
-13. What does HTTP stand for? 
-14. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?
-15. What's an ORM?
-16. What's the most commonly used ORM?
-17. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.
-18. What's a migration? 
-19. When you create a migration, does it automatically modify your database?
-20. How does a model relate to a database?
+  The initial GET request returns the restaurant ERB index page
+
+  ```ruby
+  get '/restaurants/new' do
+    erb :new
+  end
+  ```
+
+  A GET request is needed to send a user to a page to create a new restaurant
+
+  ```ruby
+  post '/restaurants' do
+    restaurant = Restaurant.new(params[:restaurant])
+    restaurant.save
+    redirect '/restaurant'
+  end
+  ```
+
+  When the new restaurant form is submitted, a POST request is sent where a new restaurant is created in the database and the user is redirected to the restaurant index page.
+
+  ```ruby
+  get '/restaurant/:id' do
+    @restaurant = Restaurant.find(params[:id])
+    erb :show
+  end
+  ```
+
+  When navigating to a specific restaurant, a GET request displays the restaurant info.
+
+  ```ruby
+  get '/restaurant/:id/edit' do
+    @restaurant = Restaurant.find(params[:id])
+    erb :edit
+  end
+  ```
+
+  When clicking an edit button or link, this GET request brings the user to an edit page to change its attributes
+
+  ```ruby
+  put '/restaurant/:id' do |id|
+    Restaurant.update(id.to_i, params[:restaurant])
+    redirect "/restaurant/#{id}"
+  end
+  ```
+
+  PUT requests manipulate the data & update ths page
+
+  ```ruby
+  delete '/restaurant/:id' do |id|
+    Restaurant.destroy(id.to_i)
+    redirect '/restaurant'
+  end
+  ```
+
+  DELETE requests delete data submitted from a button/link
+
+17. What's a migration?
+
+  A migration manipulates a table. It changes the database scheme file.
+
+18. When you create a migration, does it automatically modify your database?
+
+  Nope. You have to run db:migrate
+
+19. How does a model relate to a database?
+
+  A model stores the methods and attributes of a table-turned-object.

```

Comment:
we're basically wrapping our database in a class in a sense so that we can easily interact with our database and write our own custom methods 👍